checker/jdk/nullness/src/java/lang/ClassLoader.java

```diff
@@ -679,6 +683,8 @@ private String defineClassSourceLocation(ProtectionDomain pd)
         return source;
     }
 
+    @SuppressWarnings("nullness:dereference.of.nullable") // pd.getCodeSource is de-referenced only
```

Comment:
The real issue here is that `ProtectionDomain.getCodeSource()` is not annotated as `@Deterministic`.  That would solve the problem and is better way to resolve it.  Since `ProtectionDomain.java` is in the annotated JDK, could you make that change and remove this `@SuppressWarnings`?